verification/cbmc/stubs/memcmp_override.c

```diff
@@ -10,8 +10,8 @@
 int memcmp(const void *s1, const void *s2, size_t n) {
     //  __CPROVER_HIDE:;
     int res = 0;
-    __CPROVER_precondition(__CPROVER_r_ok(s1, n), "memcmp region 1 readable");
-    __CPROVER_precondition(__CPROVER_r_ok(s2, n), "memcpy region 2 readable");
+    __CPROVER_precondition(s1 != NULL && (n == 0 || __CPROVER_r_ok(s1, n)), "memcmp region 1 readable");
+    __CPROVER_precondition(s2 != NULL && (n == 0 || __CPROVER_r_ok(s2, n)), "memcpy region 2 readable");
```

Comment:
Could you add a comment explaining why do you need to add the extra constrains?

Reply:
Added a comment. Please check.